addons/stock/product.py

```diff
@@ -278,6 +278,7 @@ def _compute_nbr_reordering_rules(self, cr, uid, ids, field_names=None, arg=None
         'nbr_reordering_rules': fields.function(_compute_nbr_reordering_rules, string='Reordering Rules', type='integer', multi=True),
         'reordering_min_qty': fields.function(_compute_nbr_reordering_rules, type='float', multi=True),
         'reordering_max_qty': fields.function(_compute_nbr_reordering_rules, type='float', multi=True),
+        'category_id': fields.related('product_tmpl_id', 'categ_id', relation='product.category', type='many2one', string='Product Category', store=True),
```

Comment:
actually we need category_id field in group by options in product.product view.
and categ_id field is added in product.product model as related field without store attribute.  (see: https://github.com/odoo/odoo/blob/master/openerp/models.py#L3052).

and In search widget, discard all fields which has not store attribute value in group by. (see: https://github.com/odoo/odoo/blob/master/addons/web/static/src/js/views/search_menus.js#L430)

so either we have to add related field with store in product.product or need to fix something in framework or search widget to allow inherits fields in group by options.

may be change store attribute value in fields_get method if field is inherits.